lib/child_process.js

```diff
@@ -111,29 +112,25 @@ const MAX_BUFFER = 1024 * 1024;
  *   }} [options]
  * @returns {ChildProcess}
  */
-function fork(modulePath /* , args, options */) {
+function fork(modulePath, args = [], options) {
   modulePath = getValidatedPath(modulePath, 'modulePath');
 
   // Get options and args arguments.
   let execArgv;
-  let options = {};
-  let args = [];
-  let pos = 1;
-  if (pos < arguments.length && ArrayIsArray(arguments[pos])) {
-    args = arguments[pos++];
-  }
 
-  if (pos < arguments.length && arguments[pos] == null) {
-    pos++;
+  if (args == null) {
+    args = [];
+  } else if (typeof args !== 'object') {
+    throw new ERR_INVALID_ARG_VALUE('args', args);
+  } else if (!ArrayIsArray(args)) {
+    options = args;
+    args = [];
```

Comment:
Any reason not to use a validator here?
```suggestion
  } else if (typeof args === 'object' && !ArrayIsArray(args)) {
    options = args;
    args = [];
  } else {
    validateArray(args, 'args');
```

Reply:
Good idea. Done. That also updates the error thrown to ERR_INVALID_ARG_TYPE, which is what this breaking change is all about basically anyway.